presto-hive/src/main/java/com/facebook/presto/hive/metastore/SemiTransactionalHiveMetastore.java

```diff
@@ -427,7 +428,7 @@ public synchronized void finishInsertIntoExistingTable(
             String databaseName,
             String tableName,
             Path currentLocation,
-            List<String> fileNames,
+            List<StagingFileInfo> stagingFileInfos,
```

Comment:
For `finishInsertIntoExistingTable` we shouldn't be doing rename twice. It should simply accept the list of committed staging files.